syntax = "proto3";

package te_rpc;

service TEDataProvider {
  rpc ActiveTasks (InfoType) returns (stream Task) {}
}

message InfoType {
  enum Type {
    SHORT = 0;
    FULL = 1;
  }
  enum Slice {
    ALL = 0;
    NEW = 1;
  }
  Type type = 1;
  Slice slice = 2;
}

message Task {
  uint32 pid = 1;
  string name = 2;
  string path_name = 3;
  repeated string args = 4;
  map<string, string> envs = 5;
  float cpu_usage_user = 6;
  float cpu_usage_kernel = 7;
  uint64 real_mem_size = 8;
  uint64 virtual_mem_size = 9;
}

